### PR TITLE
fix docker build opts error

### LINKED ISSUE #85277: Make quick-release-images error
**What happened**:
When i do some command like :
```
KUBE_BUILD_HYPERKUBE=n \
KUBE_BUILD_CONFORMANCE=n \
KUBE_BUILD_PULL_LATEST_IMAGES=n \
 make quick-release-images 
```
An error will happen due to `KUBE_BUILD_PULL_LATEST_IMAGES=n`:
```
k8s.io/kubernetes/build/lib/release.sh:348
DOCKER_BUILD_OPTS[@]: unbound variable
```

**What you expected to happen**:
The sell is excuted normally.

**How to reproduce it (as minimally and precisely as possible)**:

Set `KUBE_BUILD_PULL_LATEST_IMAGES=n ` and run `make quick-release-images`
**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by zhouya0**
**What type of PR is this?**

> /kind bug

**What this PR does / why we need it**:
Fix the `unbound variable` error.
It's because of the `set -u` of the script. When the `DOCKER_BUILD_OPTS` is empty. It will report this error. 

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubernetes/kubernetes/issues/85277

**Does this PR introduce a user-facing change?**:

```release-note
fix docker build opts error
```

### CODE REVIEW

File: build/lib/release.sh [hunk 1]
```diff
@@ -344,6 +344,8 @@ function kube::release::create_docker_images_for_server() {
     local DOCKER_BUILD_OPTS=()
     if [[ "${KUBE_BUILD_PULL_LATEST_IMAGES}" =~ [yY] ]]; then
         DOCKER_BUILD_OPTS+=("--pull")
+    else
+        DOCKER_BUILD_OPTS+=("")
```
Comment (cblecker) [inline build/lib/release.sh]: This should not be needed, as `DOCKER_BUILD_OPTS` is defined on L344

**Review by cblecker — CHANGES_REQUESTED** [review]
/hold

File: build/lib/release.sh (hunk 1 above)
Comment (zhouya0) [reply build/lib/release.sh]: Thanks for your reply. But I tested multiple-times and i still  get the `unbound variable` error if not with `DOCKER_BUILD_OPTS+=("")`.

Comment (BenTheElder) [reply build/lib/release.sh]: https://stackoverflow.com/a/7577209

It depends on the bash version

Comment (BenTheElder) [reply build/lib/release.sh]: instead we should probably expand DOCKER_BUILD_OPTS in a way that doesn't cause this error

**Pull request closed without merging**